### PR TITLE
Documentation pass on coders/byte-*.js and squeak/*.js files

### PR DESCRIPTION
**Pull request by mzgoddard**
Live version: https://mzgoddard.github.io/scratch-sb1-converter/docs-1/docs/

Progress in documenting some of the core files in this repo.

### CODE REVIEW

File: src/coders/byte-packets.js [hunk 1]
```diff
@@ -1,9 +1,56 @@
+/**
+ * @typedef {function} PacketConstructor
+ */
+
+/**
+ * A packet of bytes represented with getter/setter properties for decoding and
+ * encoding values at given offsets mapped to names.
```
Comment (kchadha) [inline src/coders/byte-packets.js:7]: This sentence is a little confusing. What is "mapped to names"? If it's the `values` maybe rephrase to:

```suggestion
  * encoding values mapped to names, located at given offsets.
```

Comment (kchadha) [reply src/coders/byte-packets.js:7]: OR:

```suggestion
  * encoding values at given offsets as names.
```

File: src/coders/byte-primitives.js [hunk 2]
```diff
@@ -4,14 +4,66 @@ const notImplemented = () => {
...
+ * @param {Uint8Array} uint8a
+ * @param {number} position
+ * @param {*} value
+ */
+
+/**
+ * An interface for reading and writing binary values to typed arrays.
+ *
+ * Combined with {@link Packet Packet} this makes
+ * reading and writing packets of binary values easy to read and do.
```
Comment (kchadha) [inline src/coders/byte-primitives.js:45]:
```suggestion
  * reading and writing packets of binary values easy.
```

File: src/coders/byte-primitives.js [hunk 3]
```diff
@@ -4,14 +4,66 @@ const notImplemented = () => {
...
+ */
 class BytePrimitive {
+    /**
+     * @constructor
+     * @param {object} options - Options initializing BytePrimitive instance
```
Comment (kchadha) [inline src/coders/byte-primitives.js:50]:
```suggestion
      * @param {object} options - Options to initialize BytePrimitive instance
```

File: src/coders/byte-stream.js [hunk 4]
```diff
@@ -1,71 +1,137 @@
...
 
-            this.uint8 = new Uint8Array(this.buffer);
-            this.uint8.set(uint8);
+            this.uint8a = new Uint8Array(this.buffer);
+            this.uint8a.set(uint8a);
         }
     }
 
+    /**
+     * Write a value to the stream with a BytePrimitive defining how to do so
```
Comment (kchadha) [inline src/coders/byte-stream.js:81]:
```suggestion
      * Write a value to the stream (with a BytePrimitive defining how to do so)
```

**Comment by mzgoddard** [comment]
I force pushed some changes bringing in @kchadha's suggestions. But at this time this PR does not seem to be reflecting that while https://github.com/mzgoddard/scratch-sb1-converter/tree/docs is ...

File: src/coders/byte-stream.js [hunk 5]
```diff
@@ -1,71 +1,137 @@
...
         this.position += StructType.size;
         return st;
     }
 
+    /**
+     * Write bytes from the given Uint8Array array to the stream and increment
+     * the position.
+     * @param {Uint8Array} bytes - Bytes to write to the stream.
+     * @param {number=} [start=0] - Where in bytes to start writing from.
+     * @param {nubmmer=} [end=bytes.length] - Where in bytes to stop writing.
```
Comment (kchadha) [inline src/coders/byte-stream.js:125]: Maybe add whether bytes[end] will be written to or not. It looks like not..

Also there is a typo: `nubmmer` => `number`

```suggestion
     * @param {number=} [end=bytes.length] - Where in bytes to stop writing, excluding position at bytes[end].
```

File: src/squeak/byte-take-iterator.js [hunk 6]
```diff
@@ -1,13 +1,35 @@
+/**
+ * Let the passed iterator only take bytes up to a certain position.
```
Comment (kchadha) [inline src/squeak/byte-take-iterator.js:2]: This is a weird way to start a comment describing a class. Maybe:

```suggestion
 * An iterator that only takes bytes up to a certain position.
```

Comment (kchadha) [reply src/squeak/byte-take-iterator.js:2]: Alternatively, maybe just remove this line completely. The comment below makes sense without it.

File: src/squeak/field-iterator.js [hunk 7]
```diff
@@ -7,7 +7,20 @@ import {
...
 
+/**
+ * Consume values for the byte stream with a iterator-like interface.
+ */
 class Consumer {
+    /**
+     * @param {object} options - Define the consumer.
+     * @param {function} [options.type=Value] - The {@link Field} type to
+     * create.
+     * @param {BytePrimitive} options.read - How to read the third Field
```
Comment (kchadha) [inline src/squeak/field-iterator.js:18]: Maybe explain why the *third* Field argument is the special one?

File: src/squeak/fields.js [hunk 8]
```diff
@@ -38,16 +75,49 @@ class Value extends Field {
     }
 }
 
+/**
+ * A header for a FieldObject representing its class and how many fields are in
+ * the object.
+ *
+ * The Fields following are header up to its size should be collected into a
```
Comment (kchadha) [inline src/squeak/fields.js:82]: This sentence is a little confusing...

File: src/squeak/ids.js [hunk 9]
```diff
@@ -1,50 +1,181 @@
...
-    // STATIC_STRING to differentiate it from STRING in this codebase.
+
+    /** A variable field header. The next byte is the version of this object.
+     * The following byte is the number of fields in this object.
+     *
+     * In Scratch 2 this is called String. To reduce confusion in the set of
+     * types, this is called STATIC_STRING in this converter. */
     STATIC_STRING: 105,
+
+    /** A variable field header. The next byte is the version of this object.
```
Comment (kchadha) [inline src/squeak/ids.js:126]: Is this one and are the ones below done or are they TODOs?

File: src/squeak/ids.js [hunk 10]
```diff
@@ -1,50 +1,181 @@
...
+     * The following byte is the number of fields in this object. */
     WATCHER_SLIDER: 174,
+
+    /** A variable field header. The next byte is the version of this object.
+     * The following byte is the number of fields in this object. */
     LIST_WATCHER: 175
 };
 
+/**
+ * A inverted map of TYPES. Map ids numbers to their string names.
```
Comment (kchadha) [inline src/squeak/ids.js:176]:
```suggestion
  * A inverted map of TYPES. Map id numbers to their string names.
```

**Review by kchadha — COMMENTED** [review]
@mzgoddard, thanks for all the documentation! Made some minor comments/suggestions.

**Comment by mzgoddard** [comment]
Continuing in #18.
